statsmodels/tsa/arima/estimators/hannan_rissanen.py

```diff
@@ -412,6 +418,12 @@ def _stitch_fixed_and_free_params(fixed_ar_or_ma_lags, fixed_ar_or_ma_params,
 
     all_lags = np.r_[fixed_ar_or_ma_lags, free_ar_or_ma_lags]
     all_params = np.r_[fixed_ar_or_ma_params, free_ar_or_ma_params]
+    assert set(all_lags) == set(spec_ar_or_ma_lags)
 
-    sorted_idx = all_lags.argsort()
-    return list(all_params[sorted_idx])
+    lag_to_param_map = dict(zip(all_lags, all_params))
+
+    # Sort params by the order of their corresponding lags in
+    # spec_ar_or_ma_lags (e.g. SARIMAXSpecification.ar_lags or
+    # SARIMAXSpecification.ma_lags)
+    all_params_sorted = [lag_to_param_map[lag] for lag in spec_ar_or_ma_lags]
```

Comment:
#7497 originally sorted params by the order of the corresponding lags, which for now should match the order of e.g. `SARIMAXSpecification.ar_lags` (judging from `standardize_lag_order`) and thus `SARIMAXParams.ar_names`. 

However, to make the ordering more robust, the correct way is to sort params by the order of the corresponding lags inside e.g. `SARIMAXSpecification.ar_lags` (and similarly `SARIMAXSpecification.ma_lags`) in case the implementation of the ordering changes in the future.